### PR TITLE
allow booking_id and iso code in requested_equipment_group to be null

### PR DESCRIPTION
**Pull request by FinnNielsen**

### CODE REVIEW

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 1]
```diff
@@ -345,9 +345,9 @@ CREATE TABLE dcsa_im_v3_0.active_reefer_settings (
 DROP TABLE IF EXISTS dcsa_im_v3_0.requested_equipment_group CASCADE;
 CREATE TABLE dcsa_im_v3_0.requested_equipment_group (
     id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
-    booking_id uuid NOT NULL REFERENCES dcsa_im_v3_0.booking (id),
+    booking_id uuid NULL REFERENCES dcsa_im_v3_0.booking (id),
     shipment_id uuid NULL REFERENCES dcsa_im_v3_0.shipment (id),
-    requested_equipment_iso_equipment_code varchar(4) NOT NULL REFERENCES dcsa_im_v3_0.iso_equipment_code (iso_equipment_code),
+    requested_equipment_iso_equipment_code varchar(4) NULL REFERENCES dcsa_im_v3_0.iso_equipment_code (iso_equipment_code),
     requested_equipment_units integer NOT NULL,
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:351]: Wouldn't this also require a NULL?

**Comment by HenrikHL** [comment]
I guess this is needed when the Carrier provides containers (confirmed equipment) that differ from what the Shipper has requested (requested equipment). In this case the bookingID will be Null and the ShipmentID will contain a link? (same with requested equipment vs confirmed equipment)?

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql (hunk 1 above)
Comment (FinnNielsen) [reply datamodel/initdb.d/03_dcsa_im_v3_0.sql:351]: Nah shippingInstruction always has one equipment, so I hardcode it to 1 in the code

Comment (HenrikHL) [reply datamodel/initdb.d/03_dcsa_im_v3_0.sql:351]: There is a significant difference between requestedUnits and ConfirmedUnits - The ShippingInstruction must be based in ConfirmedUnits...
When the Booking is confirmed - the confirmedUnits is what is available for the SI

Comment (FinnNielsen) [reply datamodel/initdb.d/03_dcsa_im_v3_0.sql]: Al right, fixed

**Review by HenrikHL — APPROVED** [review]

**Pull request merged**